src/template.ts

```diff
@@ -1,43 +1,54 @@
-import { FrontMatterCache } from 'obsidian';
+import { getRandomHex } from "./generators";
+import { FrontMatterCache } from "obsidian";
 
-const dateTmplRegex = /{{DATE:([^}]+)}}/gm
-const frontmatterTmplRegex = /{{frontmatter:([^}]+)}}/gm
+const dateTmplRegex = /{{DATE:([^}]+)}}/gm;
+const frontmatterTmplRegex = /{{frontmatter:([^}]+)}}/gm;
 
 const replaceDateVar = (s: string, date: moment.Moment): string => {
-	const m = dateTmplRegex.exec(s)
-	if (!m) return s
-	return s.replace(m[0], date.format(m[1]))
-}
+	const m = dateTmplRegex.exec(s);
+	if (!m) return s;
+	return s.replace(m[0], date.format(m[1]));
+};
 
-const replaceFrontmatterVar = (s: string, frontmatter?: FrontMatterCache): string => {
-	if (!frontmatter) return s
-	const m = frontmatterTmplRegex.exec(s)
-	if (!m) return s
-	return s.replace(m[0], frontmatter[m[1]] || '')
-}
+const replaceFrontmatterVar = (
+	s: string,
+	frontmatter?: FrontMatterCache
+): string => {
+	if (!frontmatter) return s;
+	const m = frontmatterTmplRegex.exec(s);
+	if (!m) return s;
+	return s.replace(m[0], frontmatter[m[1]] || "");
+};
 
 interface TemplateData {
-	imageNameKey: string
-	fileName: string
-	dirName: string
-	firstHeading: string
+	imageNameKey: string;
+	fileName: string;
+	dirName: string;
+	firstHeading: string;
 }
 
-export const renderTemplate = (tmpl: string, data: TemplateData, frontmatter?: FrontMatterCache) => {
-	const now = window.moment()
-	let text = tmpl
-	let newtext
+export const renderTemplate = (
+	tmpl: string,
+	data: TemplateData,
+	frontmatter?: FrontMatterCache
+) => {
+	const now = window.moment();
+	let text = tmpl;
+	let newtext;
 	while ((newtext = replaceDateVar(text, now)) != text) {
-		text = newtext
+		text = newtext;
 	}
 	while ((newtext = replaceFrontmatterVar(text, frontmatter)) != text) {
-		text = newtext
+		text = newtext;
 	}
 
 	text = text
 		.replace(/{{imageNameKey}}/gm, data.imageNameKey)
 		.replace(/{{fileName}}/gm, data.fileName)
+		.replace(/{{randomHex:([0-9]+)}}/gm, (s, n) => {
```

Comment:
<img width="353" alt="image" src="https://user-images.githubusercontent.com/31657298/215252050-de3b9f51-2e39-48f3-b479-0d7d3a142e07.png">

@reorx Can you take a look at this part? Unfortunately, since Obsidian does not allow colon characters as the intermediary file name to apply the same convention, we would have to make significant changes to other parts of the file. Any suggestions? What if we go `{{randomHex-6}}?

Reply:
Interesting, I checked out your PR locally and cannot reproduce this error. The `{{randomHex:6}}` worked just as expected.
![image](https://user-images.githubusercontent.com/405972/215268425-f6106fb7-fa6e-4d78-94fa-efb73eb76f28.png)

Reply:
What do you see in the renaming modal? You can turn off "Auto rename" to make it appear.
![image](https://user-images.githubusercontent.com/405972/215268573-dea6031f-ee2d-47da-bee1-2853717c5409.png)

Reply:
I'm currently away from my computer; I think the auto-renaming part is the problem. When you use the renaming modal, you first calculate the final name and then rename once, but with auto-renaming, you change the filename to `{{randomHex:6}}` first and rename it again, causing the problem.

Reply:
> you change the filename to {{randomHex:6}} first and rename it again

This is not correct, even if `autoRename` is true, the actual renaming action to the file-system is still executed once.

```
		const { stem, newName, isMeaningful }= this.generateNewName(file, activeFile)
		debugLog('generated newName:', newName, isMeaningful)

		if (!isMeaningful || !autoRename) {
			this.openRenameModal(file, isMeaningful ? stem : '', activeFile.path)
			return
		}
		this.renameFile(file, newName, activeFile.path, true)
```

You can see what this line prints `debugLog('generated newName:'`, I confirmed that the `newName` here was already the expected name with generated hex string.

Reply:
Hi @anaclumos, I think this PR works. I'll merge it If you don't have time to confirm the problem.

Reply:
Hey, Just chime in, is it merged or abandoned?